velox/common/file/File.cpp

```diff
@@ -159,38 +158,45 @@ LocalWriteFile::LocalWriteFile(std::string_view path) {
   memcpy(buf.get(), path.data(), path.size());
   {
     FILE* exists = fopen(buf.get(), "rb");
-    if (exists != nullptr) {
-      throw std::runtime_error(fmt::format(
-          "Failure in LocalWriteFile: path '{}' already exists.", path));
-    }
+    VELOX_CHECK(
+        !exists, "Failure in LocalWriteFile: path '{}' already exists.", path);
   }
   file_ = fopen(buf.get(), "ab");
-  if (file_ == nullptr) {
-    throw std::runtime_error("fread failure in LocalWriteFile constructor.");
-  }
+  VELOX_CHECK(file_, "fread failure in LocalWriteFile constructor, {}.", path);
```

Comment:
nit: `file_ != nullptr`

Reply:
This basically translates to below, right?

if (UNLIKELY(!file_)) {
  // throw
}